lib/streamlit/elements/metric.py

```diff
@@ -105,7 +105,7 @@ def metric(self, label, value, delta=None, delta_color="normal"):
     def parse_label(self, label):
         if not isinstance(label, str):
             raise TypeError(
-                f"'{str(label)}' is not an accepted type. label only accepts: str"
+                f"'{str(label)}' is of type {str(type(label))}, which is not an accepted type. label only accepts: str. Please convert the label to an accepted type."
```

Comment:
nit: would be good to split this long string into multiple lines (similarly to the other `TypeError`s raised by this element)